### PR TITLE
Implementing Media Service

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Description:

This is the first iteration over the WordPressComKit MediaService. For now, based on AlamoFire, but we're aiming at switching over 100% to NSURLSession.

To test: please, run the `MediaServiceTests` Unit Tests.
This is a requirement to implement [WPiOS + Share Extension Attachments](https://github.com/wordpress-mobile/WordPress-iOS/issues/4910).

Needs Review: @astralbodies 
Thanks in advance!

### CODE REVIEW

File: WordPressComKit/RequestRouter.swift [hunk 1]
```diff
@@ -7,6 +7,7 @@ public enum RequestRouter: URLRequestConvertible {
     public static var bearerToken = ""
 
     case Me()
+    case Media(siteID: Int)
```
Comment (astralbodies) [inline WordPressComKit/RequestRouter.swift]: Lets name this `MediaNew` to be consistent with `PostNew`.

**Comment by astralbodies** [comment]
Just to be complete on the unit tests maybe validate that mapJSONToMedia is operating correctly via checking all the mapped data on the created Media object?

**Comment by astralbodies** [comment]
:shipit: After adding a unit test and fixing the name in the Router

**Comment by jleandroperez** [comment]
Thanks a lot for the review @astralbodies!!

**Pull request merged**